extensions/blocks/dialogue/components/time-stamp-control.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	BaseControl,
+	Dropdown,
+	Button,
+	__experimentalNumberControl as NumberControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+function validateValue( val, max ) {
+	return Math.max( 0, Math.min( val, max ) );
+}
+
+const timeStampMap = [ 'hour', 'min', 'sec' ];
+
+/**
+ * Helper function to parse and stringify the time stamp,
+ * in the HH:MM:SS format.
+ * `HH` is optional, meaning when it's empty,
+ * the string won't contain it, returning MM:SS.
+ *
+ * @param {object} typeValue - { type: value } pair value of the time stamp.
+ * @param {Array} smh - current [ second, menute, hour ] array values.
+ * @returns {string} HH:MM:SS format.
+ */
+function setTimeStampValue( typeValue, smh ) {
+	if ( smh.length <= 2 ) {
+		smh.unshift( '00' );
+	}
+
+	const type = Object.keys( typeValue )?.[ 0 ];
+	if ( ! type ) {
+		return smh.join( ':' );
+	}
+
+	let newValue = String( validateValue( typeValue[ type ], type === 'hour' ? 23 : 59 ) );
+
+	// Mask HH:MM:SS values.
+	if ( newValue?.length === 1 ) {
+		newValue = `0${ newValue }`;
+	} else if ( newValue?.length === 0 ) {
+		newValue = '00';
+	}
+
+	smh[ timeStampMap.indexOf( type ) ] = newValue;
+
+	// Remove HH when zero.
+	if ( smh.length === 3 && smh[ 0 ] === '00' ) {
+		smh.shift();
+	}
+
+	return smh.join( ':' );
+}
+
+function TimeStamp( { value, className, onChange, shortLabel = false } ) {
+	const smh = value.split( ':' );
+	if ( smh.length <= 2 ) {
+		smh.unshift( '00' );
+	}
+
+	return (
+		<div className={ `${ className }__timestamp-controls` }>
+			<NumberControl
+				className={ `${ className }__timestamp-control__hour` }
+				label={ shortLabel ? __( 'Hour', 'jetpack' ) : __( 'Hour', 'jetpack' ) }
+				value={ smh[ 0 ] }
+				min={ 0 }
+				max={ 23 }
+				onChange={ ( hour ) => {
+					onChange( setTimeStampValue( { hour }, smh ) );
+				} }
+			/>
+
+			<NumberControl
+				className={ `${ className }__timestamp-control__minute` }
+				label={ shortLabel ? __( 'Min', 'jetpack' ) : __( 'Minute', 'jetpack' ) }
+				value={ smh[ 1 ] }
+				min={ 0 }
+				max={ 59 }
+				onChange={ ( min ) => {
+					onChange( setTimeStampValue( { min }, smh ) );
+				} }
+			/>
+
+			<NumberControl
+				className={ `${ className }__timestamp-control__second` }
+				label={ shortLabel ? __( 'Sec', 'jetpack' ) : __( 'Second', 'jetpack' ) }
+				value={ smh[ 2 ] }
+				min={ 0 }
+				max={ 59 }
+				onChange={ ( sec ) => {
+					onChange( setTimeStampValue( { sec }, smh ) );
+				} }
+			/>
```

Comment:
I could see see some type of slider control really useful here, in addition to the number controls, to quickly dial in a time stamp value by dragging, the same way you drag the audio or video player to jump to a specific time.

(But let's save exploring that for a separate PR)

<img width="270" alt="image" src="https://user-images.githubusercontent.com/1699996/101436812-81fd1080-38d4-11eb-8ae1-1c128d1ecf91.png">

Reply:
Definitelly a Range icon is something that we'd like to add. 💯